### PR TITLE
custom webhook endpoint

### PR DESCRIPTION
**Pull request by eliassebastian**
custom webhook endpoint docs

### CODE REVIEW

File: sources/server-integrations/webhook.mdx [hunk 1]
```diff
@@ -13,17 +13,23 @@ description: "Connect your Server-side data to Converge"
...
+        6. Ensure the Webhook Token is included as a header using the header key `X-Webhook-Token`.
+        5. Verify that your Webhook Source is working by sending a dummy [Placed Order](/sources/converge-spec#placed-order) event request. Ensure the request meets the following parameters outlined in [Manually instrumenting events](/sources/server-integrations/webhook#manually-instrumenting-events)
          <Accordion icon="code" title="Code example"> 
-         <Note> You should fill in your own webhook endpoint from the Converge UI in the code example below</Note>
+         <Note> You should fill in your own webhook endpoint and webhook token from the Converge UI in the code example below</Note>
    <CodeGroup>
    
    ```cURL cURL
   curl -X POST \
   "{YOUR_POSTBACK_ENDPOINT}" \
```
Comment (thomashlvt) [inline sources/server-integrations/webhook.mdx:30]: should be updated, this is now static

File: sources/server-integrations/webhook.mdx [hunk 2]
```diff
@@ -86,15 +92,19 @@ description: "Connect your Server-side data to Converge"
...
 The webhook body has the following parameters: 
 
-- `event_name`: The name of the event
-- `event_id`: The ID of the event, used for [deduplication](/sources/source-concepts#deduplication)
-- `properties`: The event properties you want to pass, covering the Converge Spec and possibly your own custom properties.
-- `profile_properties`: The [profile properties](/sources/source-concepts#profile-properties)
-- `aliases`: Any [aliases](/sources/source-concepts#aliases)
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+| `event_name`      | `string` | **Required**. The name of the event |
```
Comment (thomashlvt) [inline sources/server-integrations/webhook.mdx:101]: @janhenriklambrechts do note that we're making `properties` `profile_properties` and `event_id` required. I'm a fan of this, such that people bump into these options when calling the converge endpoint. They'll have to explicitly put empty values to send an event, but lmk if you feel strongly.

File: sources/server-integrations/webhook.mdx [hunk 3]
```diff
@@ -86,15 +92,19 @@ description: "Connect your Server-side data to Converge"
...
+| `event_id` | `string` | **Required**. The ID of the event, used for [deduplication](/sources/source-concepts#deduplication) |
+| `properties` | `object` | **Required**. The event properties you want to pass, covering the [Converge Spec](/sources/server-integrations/webhook#event-spec) and possibly your own custom properties. |
+| `profile_properties` | `object` | **Required**. The [profile properties](/sources/source-concepts#profile-properties) |
+| `aliases` | `array[string]` | **Required**. Any [aliases](/sources/source-concepts#aliases) |
```
Comment (thomashlvt) [inline sources/server-integrations/webhook.mdx:105]: note that at least one is required

File: sources/server-integrations/webhook.mdx [hunk 4]
```diff
@@ -86,15 +92,19 @@ description: "Connect your Server-side data to Converge"
...
+| `timestamp` | `timestamp` | **Optional**. The date and time when the event occurred, expressed as a UTC timestamp in ISO 8601 format (e.g., 2024-09-02T12:34:56Z). If not provided, the current UTC time will be used by default. This timestamp should be a string representing the date and time in UTC to ensure consistency across different systems.|
```
Comment (thomashlvt) [inline sources/server-integrations/webhook.mdx:106]: @eliassebastian also think it makes sense to add an explicit check that the timezone is included in the API endpoint

```suggestion
| `timestamp` | `datetime` | **Optional**. The date and time when the event occurred. If not provided, the current time will be used. The timestamp can be expressed as a string in ISO 8601 format (e.g. `"2024-09-02T12:34:56+00:00"`) or a UTC timestamp (e.g. `1725312229`).  The string representation must specify a timezone.
```

**Review by thomashlvt — APPROVED** [review]
cc @janhenriklambrechts

File: sources/server-integrations/webhook.mdx (hunk 2 above)
Comment (janhenriklambrechts) [reply sources/server-integrations/webhook.mdx]: Big fan of this too.

File: sources/server-integrations/webhook.mdx (hunk 3 above)
Comment (janhenriklambrechts) [reply sources/server-integrations/webhook.mdx]: This should be specified clearly @eliassebastian

**Review by janhenriklambrechts — APPROVED** [review]
Some style comments on my end: 
- I spun out requirement status to a separate column for easier readability
- The Note on the new Webhook token was not concise enough, cleaned that up. 

Once @thomashlvt's comments are resolved, this is GTG from my end.

File: sources/server-integrations/webhook.mdx (hunk 1 above)
Comment (eliassebastian) [reply sources/server-integrations/webhook.mdx]: changed!

File: sources/server-integrations/webhook.mdx (hunk 3 above)
Comment (eliassebastian) [reply sources/server-integrations/webhook.mdx]: Changed

File: sources/server-integrations/webhook.mdx (hunk 4 above)
Comment (eliassebastian) [reply sources/server-integrations/webhook.mdx]: added to the PR - used a validator to check if tz info is there, test included for this too

**Pull request merged**